### DISCUSSION TITLE
adding users to team via resource in multi tenant app

### DISCUSSION THREAD

**Discussion by ghost** [filamentphp/filament#12970] [Help]
### Package

Panel builder

### Package Version

3

### How can we help you?

what is the best approach for a user resource in a multi tenant app that uses the whole teams model logic

when a tenant adds a new user via a resource, how does this work when a user can join multiple teams?

what would the logic be for creating a new tenant user / adding them to an existing team?

**Comment by dissto** [comment]
If you dont need something fancy. A simple [Select](https://filamentphp.com/docs/3.x/forms/fields/select#integrating-with-an-eloquent-relationship) could be all you need as its basically just that. A belongsToMany relationship 🤔 

Or a [relation manager](https://filamentphp.com/docs/3.x/panels/resources/relation-managers#relation-managers---interactive-tables-underneath-your-resource-forms) on the user resource

Or a [tenant profile](https://filamentphp.com/docs/3.x/panels/tenancy#adding-a-tenant-profile-page)

Or x different things...it really depends on how complex you want it 😋

**Comment by leocintrao** [comment]
@kejedi did you get a solution for this? I am stuck on this. 

I am trying a Repeater on UserResource

                    Repeater::make('teams')
                    ->relationship()
                    ->schema([
                        Hidden::make('team_id')
                        ->default(??),
                        Hidden::make('user_id')
                        ->default(auth()->user()->id),

but not getting much from it

**Reply by ghost** [reply]
no, there's no real way of doing this a standardized way. there's no docs about it.

what I was thinking was, add an "invite" action to the user resource which sends an email to a person

- if they aren't registered, it asks them to register an account, then adds them to that team
- if they are registered but not logged in, it asks them to login, then adds them to that team
- if they are registered and logged in, it adds them to that team

after any of these situations happen, it then takes them to that team dashboard

I also thought about just having a "create" action in the user resource, but this might give too much power to tenants:

- if a user already exists with that email, they will know they belong to another tenant
- if a user doesn't exist with that email, you're giving them the power to create accounts for anyone

**Reply by tahinuralam** [reply]
You can get only the current tenant id
